src/phoneField/index.tsx

```diff
@@ -0,0 +1,173 @@
+  mappedCountryPhoneDatas: mappedCountryPhoneDatas[]
+  ): mappedCountryPhoneDatas[] => mappedCountryPhoneDatas.sort(
+    (a, b) => {
+      if(a.label < b.label) return -1
+      if(a.label > b.label) return 1
+      return 0
+    }
+  )
+
+const iso2toDialCode = (iso2Value: string) => {
+  if(isEmpty(iso2Value)) {
+    return ''
+  }
+  const { dialCode } = findIso2(iso2Value)
+  return `+${dialCode}`
+}
+
+export default class PhoneField extends PureComponent<PhoneFieldProps, PhoneFieldState> {
+
+  static defaultProps: Partial<PhoneFieldProps> = {
+    defaultPhoneCountryCodeValue: '',
+    defaultphoneNumberValue: '',
+    countriesWhiteList: [],
+    customCountriesName: {},
+    isInline: true,
+  }
+
+  state = {
+    countriesDatas: PhoneField.countriesDatas(
+      this.props.countriesWhiteList, this.props.customCountriesName
+    ),
+    countriesWhiteList: this.props.countriesWhiteList,
+    phoneNumber: this.props.defaultphoneNumberValue,
+    phoneCountryCode: iso2toDialCode(this.props.defaultPhoneCountryCodeValue),
+    completePhoneNumber:
+    iso2toDialCode(this.props.defaultPhoneCountryCodeValue) + this.props.defaultphoneNumberValue,
+  }
+
+  onChange = ({ name, value }: OnChangeParameters) => {
+    this.setState({[name]: value}, () => {
+      this.props.onChange({
+        name: this.props.name,
+        value: {
+          phoneNumber: this.state.phoneNumber,
+          phoneCountryCode: this.state.phoneCountryCode,
+          completePhoneNumber: this.state.phoneCountryCode + this.state.phoneNumber,
+        }
+      })
+    })
+  }
+
+  static countriesDatas(whitelist: string[], countriesName: customCountriesName) {
+    if (!isEmpty(whitelist)) {
+      const whiteListCountriesPhoneDatas = filterIso2(whitelist)
+      const whitelistMapped = mapCountriesPhoneDatas(whiteListCountriesPhoneDatas, countriesName)
+
+      return sortCountriesPhoneDatas(whitelistMapped)
+    }
+    const countriesDatasMapped = mapCountriesPhoneDatas(countryTelData.allCountries, countriesName)
+    return sortCountriesPhoneDatas(countriesDatasMapped)
+  }
+
+  static getDerivedStateFromProps(props: PhoneFieldProps, state: PhoneFieldState) {
+    if (!isEqual(props.countriesWhiteList, state.countriesWhiteList)) {
+      return {
+        countriesDatas : PhoneField.countriesDatas(
+          props.countriesWhiteList, props.customCountriesName
+        ),
+        countriesWhiteList: props.countriesWhiteList,
+      }
+    }
+
+    return null
+  }
+
+  render() {
+    const {
+      id,
+      phoneCountryCodeLabel,
+      phoneNumberLabel,
+      phoneNumberPlaceholder,
+      defaultphoneNumberValue,
+    } = this.props
+
+    const baseClassName = 'kirk-phoneField'
+
+    const classNames = cc([
+      this.props.isInline ? `${baseClassName}--inline` : baseClassName,
+      this.props.className,
+    ])
+
+    return (
+      <div id={id} className={classNames}>
+        <SelectField
+          name="phoneCountryCode"
```

Comment:
you should use a variable here, for both the state and the SelectField name, since it is linked